### ISSUE TITLE
Qlib interoperability? Tough glass to fill and memory intensive

### ISSUE THREAD

**Issue by ziongrammer** [inspircd/inspircd#1820]
**Description**

We can either add a *description tag, structure, or class along with templates (c programmer) or full blown caching with feature threads such as interception prevention, clustering, or various encryption  or just debugging

Github@unidef/qlib

Qlib irc can mainly be used for heavy caching and detection, with enough math can even be used for prediction of packets with network streams, saving bandwidth and sometimes multiple levels of prediction. 

I honestly suggest a threat level in an enumerator along with a class, and try to use regex for the description tag

**Why this would be useful**

Briefly describe why the feature would be To protect us from the scum of the universe (jk)

Basically with enough hacking each feature can be a function with its own un/comprisable thread along with heavy caching and encryption

Let’s say irc.Zion.com gets intercepted, a detection algorithm can issue a fallback thread with enough syncing and timing

Also the threading and sync collect and disperse according to the sync algorithm and timings much like radar I think
This is mainly an unsafe security measure with with caching and clustering and a time server can be used to compress and save data 

I’d port it to c++ but I hate templates and virtual functions

Also check out algore.com he has an email and yah I guess he might have used irc he did support nasa

**Comment by SadieCat** [comment]
It's not very clear what you're asking for. What is this library and how exactly can it be used for with regards to InspIRCd?

**Comment by SadieCat** [comment]
>https://github.com/unidef/qlib/blob/master/legal/LICENSE

The license for this library is not compatible with the license of InspIRCd (GPLv2). It's incredibly vague but under the "FEDERAL LAW" that this license references (presumably the United States Berne Convention Implementation Act of 1988) all copyrighted works are All Rights Reserved unless otherwise explicitly specified.
